### ISSUE TITLE
Unable to build/install due to LNK1194 Error

### ISSUE THREAD

**Issue by Demiguise** [audreyt/node-webworker-threads#111]
Currently failing to build the module with a LNK1194 pointing towards 'node.exe'.
Have been so far unable to find any information that helps with it as other's are installing for global or the problem is with iojs.exe.

Error from build:
```
LINK : fatal error LNK1194: cannot delay-load 'node.exe' due to import of data symbol '"__declspec(dllimport) const v8::ArrayBuffer::Allocator::`vftable'" (__imp_??_7Allocator@ArrayBuffer@v8@@6B@)';
link without /DELAYLOAD:node.exe [<snip>\node_modules\webworker-threads\build\WebWorkerThreads.vcxproj]
```

Running Win 7, Visual Studio 2015 installed and Python 2.7 set.
Node.js is v6.3.0, NPM is v3.10.5 and node-gyp is v3.4.0 and I'm using NW.js to build the application.

From what I can tell, my node-gyp toolchain appears to be just fine as I'm able to install other modules that require compilation (BSON/Node-sass).

**Comment by Demiguise** [comment]
I've forced node-gyp to not delay hook by adding "'win_delay_load_hook': 'false'" to the binding.gyp file, which has changed the error. Now it appears that the build passes but attempting to load the module through "require('webworker-threads')" reports that it did not self-register.

**Comment by Demiguise** [comment]
Not very well versed in Node.js inner workings but I imagine that the reason the above switched over, and started to work, is that the build was linking directly to node.exe. 

However, since I'm most likely using an different instance in some way, it's not registering with the currently running node.exe as it expects. Hence the delay_load_hook. So back to square one I think.

**Comment by Demiguise** [comment]
Attempting to build another c based project, like Npool, didn't give the same error and I believe it's down to the node.exe not being delay loaded.
After some more searching I believe that delay hooking only the iojs.exe is the correct answer to getting the build process working, however the above problem for self-registering is down to nw.js. Don't believe this is a problem for this project and more likely down to my toolset/experience.

**Comment by TimTinkers** [comment]
I'm running into the same issue here. Did you ever manage to get webworker-threads installed?

**Comment by Demiguise** [comment]
I did though my exact issue was because I was using the wrong tool to compile.
If you're using Node-Webkit then I suggest you grab "nw-gyp" via npm do the following.

1. Download the full repo from here and unpack it locally.
2. Add "'win_delay_load_hook': "<!(node -e \"var v = process.version.substring(1,2); console.log(v > 0 && v < 4);\")"," to the binding.gyp file. If you build now with nw-gyp instead there should be an error to do with DecodeBytes/DecodeWrite not fitting.
3. Edit "bson.cc" and find the "NAN_METHOD(BSON::BSONDeserialize)" function. The errors, for me at least, were here.

So to fix them I added the following to that function around ~795.
```
else
{
	Isolate* isolate = Isolate::GetCurrent();
	// The length of the data for this encoding
			intptr_t len = DecodeBytes(isolate, info[0], BINARY);

	// Validate that we have at least 5 bytes
			if(len < 5) return Nan::ThrowError("corrupt bson message < 5 bytes long");

	// Let's define the buffer size
	char* data = (char *)malloc(len);
			DecodeWrite(isolate, data, len, info[0], BINARY)
}
```

Hopefully that helps.
